pkg/providers/nutanix/template.go

```diff
@@ -79,7 +83,15 @@ func (ntb *TemplateBuilder) GenerateCAPISpecWorkers(clusterSpec *cluster.Spec, w
 }
 
 func (ntb *TemplateBuilder) GenerateCAPISpecSecret(clusterSpec *cluster.Spec, buildOptions ...providers.BuildMapOption) (content []byte, err error) {
-	values := buildTemplateMapSecret(clusterSpec, ntb.creds)
+	encodedCreds, _ := json.Marshal(ntb.creds)
+	nutanixCreds := credentials.NutanixCredentials{
+		Credentials: []credentials.Credential{{
+			Type: credentials.BasicAuthCredentialType,
+			Data: encodedCreds,
+		}},
+	}
+	credsJSON, _ := json.Marshal(nutanixCreds.Credentials)
```

Comment:
Doesn't look like we need the `credentials.NutanixCredentials` struct. We can use `credentials.Credentials` directly
```suggestion
	nutanixCreds := []credentials.Credential{{
		Type: credentials.BasicAuthCredentialType,
		Data: encodedCreds,
	}},
	credsJSON, _ := json.Marshal(nutanixCreds)
```